.github/scripts/trymerge.py

```diff
@@ -735,22 +739,29 @@ def find_matching_merge_rule(pr: GitHubPR,
                                  f"{','.join(list(rule_approvers_set)[:5])}{', ...' if len(rule_approvers_set) > 5 else ''}")
             continue
         if rule.mandatory_checks_name is not None:
-            pass_checks = True
+            pending_checks = []
+            failed_checks = []
             checks = pr.get_checkrun_conclusions()
             # HACK: We don't want to skip CLA check, even when forced
             for checkname in filter(lambda x: force is False or "CLA Check" in x, rule.mandatory_checks_name):
-                if checkname not in checks or checks[checkname] != "SUCCESS":
-                    if reject_reason_score < 20000:
-                        reject_reason_score = 20000
-                        reject_reason = f"Refusing to merge as mandatory check {checkname} "
-                        reject_reason += "has not been run" if checkname not in checks else "failed"
-                        reject_reason += f" for rule {rule_name}"
-                    pass_checks = False
-            if not pass_checks:
-                continue
+                if checkname not in checks or checks[checkname]:
+                    pending_checks.append(checkname)
+                elif checks[checkname] != 'SUCCESS':
+                    failed_checks.append(checkname)
+        if len(failed_checks) > 0:
+            if reject_reason_score < 30000:
+                reject_reason_score = 30000
+                reject_reason = f"Refusing to merge as mandatory check(s) {','.join(failed_checks)} failed for rule {rule_name}"
+        elif len(pending_checks) > 0:
+            if reject_reason_score < 20000:
+                reject_reason_score = 20000
+                reject_reason = f"Refusing to merge as mandatory check(s) {','.join(pending_checks)}"
+                reject_reason += f"are pending for rule {rule_name}"
```

Comment:
Or not run at all, right?
```suggestion
                reject_reason += f"are not yet run for rule {rule_name}"
```